From b2ff04a5cb7a5aafa04725043d3c35594cc26bf7 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Thu, 2 Apr 2015 16:20:14 +0800
Subject: [PATCH] MLK-10518-2 ARM: dts: add support for sii902x hdmi audio

Add sii902x hdmi sound card.
Set clk IMX7D_PLL_AUDIO_POST_DIV to 884736000HZ.
Set clk IMX7D_SAI1_ROOT_SRC parent to IMX7D_PLL_AUDIO_POST_DIV.
Set clk IMX7D_SAI1_ROOT_CLK to 18432000HZ.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
---
 arch/arm/boot/dts/imx7d-sdb-hdmi.dts | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-sdb-hdmi.dts b/arch/arm/boot/dts/imx7d-sdb-hdmi.dts
index ac8dc38..2825e00 100644
--- a/arch/arm/boot/dts/imx7d-sdb-hdmi.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-hdmi.dts
@@ -8,7 +8,29 @@
 
 #include "imx7d-sdb.dts"
 
+/ {
+	sound-hdmi {
+		compatible = "fsl,imx7d-sdb-sii902x",
+			   "fsl,imx-audio-sii902x";
+		model = "sii902x-audio";
+		cpu-dai = <&sai1>;
+		hdmi-controler = <&sii902x>;
+	};
+};
+
+&clks {
+	assigned-clocks = <&clks IMX7D_PLL_AUDIO_POST_DIV>;
+	assigned-clock-rates = <884736000>;
+};
+
 &sii902x {
 	status = "okay";
 };
 
+&sai1 {
+	assigned-clocks = <&clks IMX7D_SAI1_ROOT_SRC>,
+			  <&clks IMX7D_SAI1_ROOT_CLK>;
+	assigned-clock-parents = <&clks IMX7D_PLL_AUDIO_POST_DIV>;
+	assigned-clock-rates = <0>, <18432000>;
+	status = "okay";
+};
-- 
1.8.0

